### PR TITLE
[stories/US331823] .NET SDK returns only 200 patients

### PR DESCRIPTION
**Pull request by lanyadasilva**
## Before requesting a reviewer, complete this checklist.

- [x] I have updated `releaseHistory.md`.
- [x] I have updated the `<Version>` attribute with the new version in `proknow-sdk.csproj` following the [semantic versioning](https://semver.org/) rules.

### CODE REVIEW

File: proknow-sdk/Collection/CollectionPatients.cs [hunk 1]
```diff
@@ -56,8 +60,8 @@ public async Task<IList<CollectionPatientSummary>> QueryAsync()
                     { "workspace", _collectionItem.WorkspaceIds[0] }
                 };
             }
-            var responseJson = await _proKnow.Requestor.GetAsync(route, null, queryParameters);
-            return DeserializeCollectionPatientSummaries(responseJson);
+            var responseJson = await _proKnow.Requestor.GetAsyncWithPaging(route, null, queryParameters);
+            return DeserializeCollectionPatientSummariesWithPaging(responseJson);
```
Comment (MohanVallivedu) [inline proknow-sdk/Collection/CollectionPatients.cs:64]: I think we can delete the DeserializeCollectionPatientSummaries method

File: proknow-sdk/Collection/CollectionPatients.cs [hunk 2]
```diff
@@ -97,5 +101,28 @@ private IList<CollectionPatientSummary> DeserializeCollectionPatientSummaries(st
             }
             return collectionPatientSummaries;
         }
+
+        /// <summary>
+        /// Creates a collection of collection patient summaries from their JSON representation
+        /// </summary>
+        /// <param name="json">JSON representation of a collection of collection patient summaries</param>
```
Comment (MohanVallivedu) [inline proknow-sdk/Collection/CollectionPatients.cs]:
```suggestion
        /// <param name="json">A list of JSON representation of a collection of collection patient summaries</param>
```

File: proknow-sdk-test/TestHelper.cs [hunk 3]
```diff
@@ -85,6 +87,73 @@ public static async Task AssemblyCleanup()
...
+
+            //// Create the patients
+            //var patientCreateTasks = new List<Task<PatientItem>>();
+            //for (int i = 0; i < numPatients; i++)
+            //{
+            //    var mrn = $"{testNumber}-{i}-Mrn";
+            //    var name = $"{testNumber}-{i}-Name";
+            //    if (i % 500 == 0)
+            //    {
+            //        await Task.Delay(1000);
```
Comment (MohanVallivedu) [inline proknow-sdk-test/TestHelper.cs]: I think both ways looks good. We can even make it to delay for every 250 patients to be more safe.
Let me know what you think @kyleburnett

**Review by MohanVallivedu — APPROVED** [review]

File: proknow-sdk-test/TestHelper.cs [hunk 4]
```diff
@@ -85,6 +87,73 @@ public static async Task AssemblyCleanup()
...
+            //    }
+            //    patientCreateTasks.Add(_proKnow.Patients.CreateAsync(workspaceItem.Id, mrn, name));
+            //}
+            //var patientItems = await Task.WhenAll(patientCreateTasks);
+            //return patientItems;
+
+            //++++++++++++++++++++++++++
+
+            //Limit number of active requests
```
Comment (kyleburnett) [inline proknow-sdk-test/TestHelper.cs]:
```suggestion
            // Limit number of active requests

```

File: proknow-sdk/Requestor.cs [hunk 5]
```diff
@@ -268,6 +284,54 @@ internal async Task<string> GetDomainStatusAsync()
...
+                    responseContent = await response.Content.ReadAsStringAsync();
+                }
+                if (response.Content.Headers.ContentType.MediaType == "text/html" && responseContent.Length >= 13 && responseContent.Substring(0, 14).ToUpper() == "<!DOCTYPE HTML")
+                {
+                    // Response content is index.html from ProKnow, most likely due to invalid base URL
+                    var baseUrlWithoutApi = _baseUrl.Substring(0, _baseUrl.Length - 4);
+                    throw new ProKnowHttpException(method.ToString(), $"{_baseUrl}{route}", HttpStatusCode.NotFound.ToString(), $"Please verify the base URL '{baseUrlWithoutApi}'.");
+                }
+				
+				//determine if paging is required and parameters needed for the next request
```
Comment (kyleburnett) [inline proknow-sdk/Requestor.cs]:
```suggestion
				// Determine if paging is required and parameters needed for the next request
```

File: proknow-sdk/Requestor.cs [hunk 6]
```diff
@@ -268,6 +284,54 @@ internal async Task<string> GetDomainStatusAsync()
...
+                if (response.Content != null)
+                {
+                    responseContent = await response.Content.ReadAsStringAsync();
+                }
+                if (response.Content.Headers.ContentType.MediaType == "text/html" && responseContent.Length >= 13 && responseContent.Substring(0, 14).ToUpper() == "<!DOCTYPE HTML")
+                {
+                    // Response content is index.html from ProKnow, most likely due to invalid base URL
+                    var baseUrlWithoutApi = _baseUrl.Substring(0, _baseUrl.Length - 4);
+                    throw new ProKnowHttpException(method.ToString(), $"{_baseUrl}{route}", HttpStatusCode.NotFound.ToString(), $"Please verify the base URL '{baseUrlWithoutApi}'.");
+                }
```
Comment (kyleburnett) [inline proknow-sdk/Requestor.cs]: I don't think we need to handle this specific case in the SDK.

**Review by kyleburnett — CHANGES_REQUESTED** [review]

File: proknow-sdk-test/TestHelper.cs (hunk 3 above)
Comment (kyleburnett) [reply proknow-sdk-test/TestHelper.cs]: I'm okay with either as well. Whichever you think would be faster as long as both are reliable.

Comment (lanyadasilva) [reply proknow-sdk-test/TestHelper.cs]: Both were reliable for me.  Time-wise, on average, both were very similar, though the semaphore version was slightly faster more often.  So I'll go with that.  Thanks!

**Review by kyleburnett — APPROVED** [review]

**Pull request merged**